units/pod_allocator.go

```diff
@@ -93,6 +113,26 @@ func (j *podAllocatorJob) Run(ctx context.Context) {
 	}
 }
 
+func (j *podAllocatorJob) canAllocate() (shouldAllocate bool, err error) {
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return false, errors.Wrap(err, "getting admin settings")
+	}
+	numInitializing, err := pod.CountByInitializing()
+	if err != nil {
+		return false, errors.Wrap(err, "counting currently-initializing pods")
+	}
+	if numInitializing >= settings.PodInit.MaxParallelPodRequests {
+		return false, nil
+	}
+
+	flags, err := evergreen.GetServiceFlags()
```

Comment:
Not really a big deal, but it feels like this check should come first since if the pod allocator is disabled, no point in doing the count query.

Reply:
Good catch, I swapped them.